From 54f923a0dc07f9c55d67f434b1ec5fbc012fb533 Mon Sep 17 00:00:00 2001
From: Jami Kettunen <jami.kettunen@protonmail.com>
Date: Wed, 13 Nov 2024 02:07:10 +0200
Subject: [PATCH 05/15] [REVISIT] Generate asteroidcontrolsplugin qrc via CMake

Re-use the already existing listing of controls used for documentation
generation as well.

Potentially merge qt_qml_add_module() migration into this as well
---
 src/controls/CMakeLists.txt | 14 +++++++++++---
 src/controls/resources.qrc  | 29 -----------------------------
 2 files changed, 11 insertions(+), 32 deletions(-)
 delete mode 100644 src/controls/resources.qrc

diff --git a/src/controls/CMakeLists.txt b/src/controls/CMakeLists.txt
index d008466..2512bd9 100644
--- a/src/controls/CMakeLists.txt
+++ b/src/controls/CMakeLists.txt
@@ -11,9 +11,7 @@ set(HEADERS
 	src/flatmeshnode.h
 	src/icon.h)
 
-add_library(asteroidcontrolsplugin ${SRC} ${HEADERS} resources.qrc)
-
-set(controls 
+set(controls
         Application
         BorderGestureArea
         CircularSpinner
@@ -40,6 +38,16 @@ set(controls
         TextBase
         TextField
 )
+foreach(control ${controls})
+    list(APPEND controls-qml "qml/${control}.qml")
+endforeach()
+
+add_library(asteroidcontrolsplugin ${SRC} ${HEADERS})
+
+qt_add_resources(asteroidcontrolsplugin "asteroidcontrolsplugin_qrc"
+    PREFIX "/org/asteroid/controls/"
+    FILES "${controls-qml}")
+
 set(controls-docs "$<LIST:TRANSFORM,$<LIST:TRANSFORM,$<LOWER_CASE:${controls}>,PREPEND,qml->,APPEND,.html>")
 set(doc-dir "${CMAKE_BINARY_DIR}/doc/html")
 set(full-controls-docs "$<LIST:TRANSFORM,${controls-docs},PREPEND,${doc-dir}>")
diff --git a/src/controls/resources.qrc b/src/controls/resources.qrc
deleted file mode 100644
index 7c19457..0000000
--- a/src/controls/resources.qrc
+++ /dev/null
@@ -1,29 +0,0 @@
-<RCC>
-    <qresource prefix="/org/asteroid/controls/">
-        <file>qml/Application.qml</file>
-        <file>qml/BorderGestureArea.qml</file>
-        <file>qml/CircularSpinner.qml</file>
-        <file>qml/Dims.qml</file>
-        <file>qml/ListItem.qml</file>
-        <file>qml/HandWritingKeyboard.qml</file>
-        <file>qml/HighlightBar.qml</file>
-        <file>qml/IconButton.qml</file>
-        <file>qml/Indicator.qml</file>
-        <file>qml/Label.qml</file>
-        <file>qml/LayerStack.qml</file>
-        <file>qml/Marquee.qml</file>
-        <file>qml/LabeledSwitch.qml</file>
-        <file>qml/LabeledActionButton.qml</file>
-        <file>qml/PageDot.qml</file>
-        <file>qml/PageHeader.qml</file>
-        <file>qml/IntSelector.qml</file>
-        <file>qml/SegmentedArc.qml</file>
-        <file>qml/SpinnerDelegate.qml</file>
-        <file>qml/Spinner.qml</file>
-        <file>qml/StatusPage.qml</file>
-        <file>qml/Switch.qml</file>
-        <file>qml/TextArea.qml</file>
-        <file>qml/TextBase.qml</file>
-        <file>qml/TextField.qml</file>
-    </qresource>
-</RCC>
-- 
2.47.0

